### DISCUSSION TITLE
Curious, does MLX allow to just port my pytorch code? I don't want to rewrite it. It takes a lot of time to write my pytorch code.

### DISCUSSION THREAD

**Discussion by brando90** [ml-explore/mlx#582] [Q&A]
Curious, does MLX allow to just port my pytorch code? I don't want to rewrite it. It takes a lot of time to write my pytorch code.

**Comment by awni** [comment]
If you are using high-level PyToch e.g. `nn` then mostly you can replace those modules with `mlx.nn`. But right now the replacement is not perfectly 1:1 so it does require some work.

**Comment by brando90** [comment]
I have hugging face code I do NOT want to rewrite. That's the best use case.
-----
Brando Miranda
Ph.D. Student
Computer Science, Stanford University
EDGE Scholar, Stanford University
***@***.***
website: https://brando90.github.io/brandomiranda/home.html
mentorship opportunities: https://brando90.github.io/brandomiranda/prospective-collaborations.html



On Jan 29, 2024, at 10:57 AM, Awni Hannun ***@***.***> wrote:



If you are using high-level PyToch e.g. nn then mostly you can replace those modules with mlx.nn. But right now the replacement is not perfectly 1:1 so it does require some work.
